pages/admin/lessons/new/index.tsx

```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react'
+import { gql, useLazyQuery } from '@apollo/client'
+import { AdminLessonCard } from '../../../../components/admin/lessons/AdminLessonCard'
+import { withGetApp, GetAppProps } from '../../../../graphql'
+import { AdminLayout } from '../../../../components/admin/AdminLayout'
+import Link from 'next/link'
+import styles from '../../../../scss/adminLessonPage.module.scss'
+
+const EXERCISES = gql`
+  query {
+    exercises {
+      flaggedAt
+      module {
+        lesson {
+          title
+        }
+      }
+    }
+  }
+`
+type Exercise = {
+  flaggedAt?: string
+  module: {
+    lesson: {
+      title: string
+    }
+  }
+}
+
+type Exercises = Exercise[]
+
+const Lessons: React.FC<GetAppProps> = ({ data }) => {
+  const { lessons } = data
+  const [lessonCards, setCards] = useState<[]>([])
+  const [getExercises, { data: exercisesData }] = useLazyQuery<{
+    exercises: Exercises
+  }>(EXERCISES, {
+    onCompleted: async () => {
+      const exerciseMapping = await exercisesData?.exercises.reduce(
+        (prev: any, cur) => {
+          if (cur.flaggedAt) {
+            if (!prev[cur.module?.lesson?.title]) {
+              prev[cur.module?.lesson?.title] = 1
+              return prev
+            }
+            prev[cur.module?.lesson?.title]++
+          }
+
+          return prev
+        },
+        {}
+      )
+
+      setCards(
+        lessons?.reduce((prev: any, cur: any, i) => {
+          prev.push(
+            <div key={i} className={styles.lessonCard}>
+              <AdminLessonCard
+                lesson={cur}
+                pendingFlaggedQuestions={
+                  exerciseMapping[cur.title] ? exerciseMapping[cur.title] : 'No'
+                }
+              />
+            </div>
+          )
+          return prev
+        }, [])
+      )
+    }
+  })
+
+  React.useEffect(() => {
+    lessons && getExercises()
```

Comment:
What is the reason this useEffect is needed?

Reply:
I'm guessing this is to handle the use case of the `lessons still loading` from getApp query. If that is the case, please try to use AdminLayout component to handle this case. It should have sufficient logic to handle this use case

Reply:
Can i get some additional guidance on how `AdminLayout` can handle this? Im already passing in the `data` that `lessons` was deconstructed from to the `data` prop in `AdminLayout`

Reply:
@HS-90 Adminlayout returns a loading spinner when data is still loading. It will only return its child elements when data is done loading

https://github.com/garageScript/c0d3-app/blob/3a064c05b09b9ec65049904ddef3457a48e4b9cd/components/admin/AdminLayout.tsx#L36

Reply:
@anthonykhoa is there an example in the codebase for this? I've looked at the AdminLayout file and see the logic, but I'm still having a hard time applying it.

Reply:
I am not sure about other instances(I haven't contributed in awhile, so i'm less familiar with the codebase),  but you can try searching `AdminLayout` and see all the files that use it. Looks like there are only 7 results, so there's not much to have to peruse through:
https://github.com/garageScript/c0d3-app/search?q=adminlayout

`it doesn't wait for the lessons data to load on a reliable basis unless we apply the useEffect`. <--
That is because this `Lesson` component is not a child element of AdminLayout, instead it just uses the component in its return statement.

What I would try doing(and optimally, I would try to think of multiple proposals on how to achieve a solution to compare pros/cons) is:

1) Create a new component `component X`(needa change name of course) to do all that is needed to do after `lessons data` is fully loaded. This way, the new component will not have to worry about lessons data not being loaded. 

2) Use component `X` as a child component of AdminLayout component
```
<AdminLayout>
  <X />
</AdminLayout>
```

With this method, you can don't have to use a lazy hook either

Reply:
Something else that might work is:
1) Create new component `component X`. Except this time, you use a codegen graphql hook `withGetExercises`(im just guessing the name, idk what was actually generated)

2) Use Component `X` as a child component of AdminLayout component
```
<AdminLayout>
  {withGetExercises()(X)}
</AdminLayout>
```

Also,I haven't made POCS for any of these proposals I gave, so you would have to test to confirm if they will work

Reply:
Alternatively, you can turn adminlayout itself into a container

example container
https://github.com/garageScript/c0d3-app/blob/master/containers/withQueryLoader.tsx

Reply:
> I am not sure about other instances(I haven't contributed in awhile, so i'm less familiar with the codebase), but you can try searching `AdminLayout` and see all the files that use it. Looks like there are only 7 results, so there's not much to have to peruse through: https://github.com/garageScript/c0d3-app/search?q=adminlayout
> 
> `it doesn't wait for the lessons data to load on a reliable basis unless we apply the useEffect`. <-- That is because this `Lesson` component is not a child element of AdminLayout, instead it just uses the component in its return statement.
> 
> What I would try doing(and optimally, I would try to think of multiple proposals on how to achieve a solution to compare pros/cons) is:
> 
> 1. Create a new component `component X`(needa change name of course) to do all that is needed to do after `lessons data` is fully loaded. This way, the new component will not have to worry about lessons data not being loaded.
> 2. Use component `X` as a child component of AdminLayout component
> 
> ```
> <AdminLayout>
>   <X />
> </AdminLayout>
> ```
> 
> With this method, you can don't have to use a lazy hook either

This method worked. It actually made my old method look a little over-complicated/over-engineered. I want to try the  other methods you suggested before updating this PR, but this was a good way to do it. Also good learning experience for me.